src/main/java/edu/kit/hci/soli/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,173 @@
+package edu.kit.hci.soli.service.impl;
+
+import edu.kit.hci.soli.domain.User;
+import edu.kit.hci.soli.repository.UserRepository;
+import edu.kit.hci.soli.service.BookingsService;
+import edu.kit.hci.soli.service.UserService;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.stereotype.Service;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Service class for managing {@link User} entities.
+ * Provides methods for user creation, retrieval, and management.
+ */
```

Comment:
If the interface is documented, the documentation of this class (mostly) becomes redundant